qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -584,10 +584,14 @@ def on_load_started(self, tab):
     @pyqtSlot()
     def on_cur_load_started(self):
         """Leave insert/hint mode when loading started."""
-        modeman.leave(self._win_id, usertypes.KeyMode.insert, 'load started',
-                      maybe=True)
-        modeman.leave(self._win_id, usertypes.KeyMode.hint, 'load started',
-                      maybe=True)
+        if config.instance.get('input.insert_mode.leave_on_load',
+                               url=self.current_url()):
```

Comment:
This seems to break when the URL is invalid - I've thought about just returning the global value if the URL is invalid, but that seems like a source of hard-to-catch bugs. Here, you'll either need to do something like:

```python
url = self.current_url()
if not url.isValid():
    url = None
```

or (since we already have something similar elsewhere), add something like `ignore_invalid_url` or so to `config.instance.get()`.